addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/handler/GlobalCacheHandler.java

```diff
@@ -788,10 +856,201 @@ private void checkConnection() {
                 markThingOnline();
                 deviceIsConnected = true;
             } else {
-                logger.trace("Connection check failed for thing {} at IP {}", thingID(), commandConnection.getIP());
+                logger.debug("Connection check failed for thing {} at IP {}", thingID(), commandConnection.getIP());
                 disconnect();
             }
         }
+
+        private void startSerialReaders() {
+            if (deviceSupportsSerial1()) {
+                serialReader1 = startSerialReader(CommandType.SERIAL1, ENABLE_TWO_WAY_1, EOM_DELIMITER_1);
+            }
+            if (deviceSupportsSerial2()) {
+                serialReader2 = startSerialReader(CommandType.SERIAL2, ENABLE_TWO_WAY_2, EOM_DELIMITER_2);
+            }
+        }
+
+        private SerialReader startSerialReader(CommandType serialDevice, String enableTwoWayConfig,
+                String eomDelimiterConfig) {
+            Boolean enableTwoWay = (Boolean) thing.getConfiguration().get(enableTwoWayConfig);
+            logger.debug("Enable two-way is {} for thing {} {}", enableTwoWay, thingID(), serialDevice);
+
+            if (enableTwoWay != null && enableTwoWay.equals(Boolean.TRUE)) {
+                // Get the end of message delimiter from the config, URL decode it, and convert it to a byte array
+                String eomString = (String) thing.getConfiguration().get(eomDelimiterConfig);
+                if (StringUtils.isNotEmpty(eomString)) {
+                    logger.debug("End of message is {} for thing {} {}", eomString, thingID(), serialDevice);
+                    byte[] eom;
+                    try {
+                        eom = URLDecoder.decode(eomString, CHARSET).getBytes(CHARSET);
+                    } catch (UnsupportedEncodingException e) {
+                        logger.info("Unable to decode end of message delimiter {} for thing {} {}", eomString,
+                                thingID(), serialDevice, e);
+                        return null;
+                    }
+
+                    // Start the serial reader using the above end-of-message delimiter
+                    SerialReader serialReader = new SerialReader(serialDevice, getSerialIn(serialDevice), eom);
+                    serialReader.start();
+                    return serialReader;
+                } else {
+                    logger.warn("End of message delimiter is not defined in configuration of thing {}", thingID());
+                }
+            }
+            return null;
+        }
+
+        private void stopSerialReaders() {
+            if (deviceSupportsSerial1() && serialReader1 != null) {
```

Comment:
1. I wanted to be consistent with how I was doing it in the method to start the readers.
2. Yes